### PR TITLE
Add nextWednesday, nextThursday, nextFriday, nextSaturday, nextSunday

### PR DESCRIPTION
**Pull request by tan75**
### New Features
nextWednesday, nextThursday, nextFriday, nextSaturday, nextSunday

### CODE REVIEW

**Comment by 0x24a537r9** [comment]
I think you probably want to run `sh scripts/build/build.sh` to rebuild the indices. I put up #2309 to fix `next(Day|Monday|Tuesday)` :)

**Comment by tan75** [comment]
> I think you probably want to run `sh scripts/build/build.sh` to rebuild the indices. When you do, you'll run into an issue where the `Day` type is not being properly created in the Flow types, so it might be a nontrivial fix

Thanks for a heads-up!
We haven't encountered it yet however if we do - we will fix it.

**Comment by kossnocorp** [comment]
@0x24a537r9 could you please elaborate? I did release `v2.19.0` with `nextModay` and `nextTuesday`, did you have any problem with that?

File: src/nextSaturday/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @param {Date | number} date the date to check
+ * @returns {Date} the date is the next Saturday
+ * @throws {TypeError} the date should be a valid date
+ * @throws {TypeError} 1 argument required
+ *
+ * @example
+ * When is the next Saturday after Mar, 22, 2020?
+ * const result = nextSaturday(new Date(2020, 2, 22))
+ * => Sat Mar 28 2020 00:00:00
+ **/
```
Comment (kossnocorp) [inline src/nextSaturday/index.ts:22]: `**/` hangs JSDoc parser, please replace it with `*/` here and everywhere (this is the correct JSDoc sequence `/** Comment */`.

File: src/nextFriday/test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      nextFriday(new Date(2020, 2 /* Mar */, 18)),
+      new Date(2020, 2 /* Mar */, 20)
+    )
+
+    assert.deepStrictEqual(
+      nextFriday(new Date(2020, 2 /* Mar */, 17)),
+      new Date(2020, 2 /* Mar */, 20)
+    )
+  })
+  it('returns `Invalid Date` if the given date is invalid', function () {
```
Comment (kossnocorp) [inline src/nextFriday/test.ts:44]: Please add space for consistency sake. Also in new tests code, we can use `() => {}`

```suggestion

  it('returns `Invalid Date` if the given date is invalid', () => {
```

File: src/nextFriday/index.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * @throws {TypeError} the date should be a valid date
+ * @throws {TypeError} 1 argument required
+ *
+ * @example
+ * When is the next Friday after Mar, 22, 2020?
+ * const result = nextFriday(new Date(2020, 2, 22))
+ * => Fri Mar 27 2020 00:00:00
+ **/
+
+export default function nextFriday(date: Date | number): Date {
```
Comment (kossnocorp) [inline src/nextFriday/index.ts:24]: Please remove space and `**/` (see my other comment about it with more details).

```suggestion
 */
export default function nextFriday(date: Date | number): Date {
```

File: src/nextSaturday/index.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+export default function nextSaturday(date: Date | number): Date {
```
Comment (kossnocorp) [inline src/nextSaturday/index.ts:24]:
```suggestion
export default function nextSaturday(date: Date | number): Date {
```

File: src/nextFriday/index.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * @name nextFriday
+ * @category Weekday Helpers
+ * @summary When is the next Friday?
+ *
+ * @description
+ * When is the next Friday?
+ *
+ * @param {Date | number} date the date to check
+ * @returns {Date} the date is the next Friday
+ * @throws {TypeError} the date should be a valid date
```
Comment (kossnocorp) [inline src/nextFriday/index.ts:15]: I don't think it's still true

```suggestion
```

File: src/nextSaturday/index.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * @name nextSaturday
+ * @category Weekday Helpers
+ * @summary When is the next Saturday?
+ *
+ * @description
+ * When is the next Saturday?
+ *
+ * @param {Date | number} date the date to check
+ * @returns {Date} the date is the next Saturday
+ * @throws {TypeError} the date should be a valid date
```
Comment (kossnocorp) [inline src/nextSaturday/index.ts:15]:
```suggestion
```

File: src/nextFriday/index.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @name nextFriday
+ * @category Weekday Helpers
+ * @summary When is the next Friday?
+ *
+ * @description
+ * When is the next Friday?
+ *
+ * @param {Date | number} date the date to check
```
Comment (kossnocorp) [inline src/nextFriday/index.ts:13]:
```suggestion
 * @param {Date | number} date - the date to start counting from
```

File: src/nextFriday/index.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * @returns {Date} the date is the next Friday
```
Comment (kossnocorp) [inline src/nextFriday/index.ts:14]:
```suggestion
 * @returns {Date} the next Friday
```

File: src/nextMonday/index.ts [hunk 9]
```diff
@@ -10,10 +10,10 @@ import toDate from '../toDate/index'
  * @description
  * When is the next Monday?
  *
- * @param {Date | number} date - the date to check
- * @returns {Date} - the date is the next Monday
- * @throws {TypeError} - the date should be a valid date
- * @throws {TypeError} - 1 argument required
+ * @param {Date | number} date the date to check
```
Comment (kossnocorp) [inline src/nextMonday/index.ts:13]:
```suggestion
 * @param {Date | number} date - the date to start counting from
```

File: src/nextMonday/index.ts [hunk 10]
```diff
@@ -10,10 +10,10 @@ import toDate from '../toDate/index'
...
+ * @returns {Date} the date is the next Monday
```
Comment (kossnocorp) [inline src/nextMonday/index.ts:14]:
```suggestion
 * @returns {Date} the next Monday
```

File: src/nextMonday/index.ts [hunk 11]
```diff
@@ -10,10 +10,10 @@ import toDate from '../toDate/index'
...
+ * @throws {TypeError} the date should be a valid date
```
Comment (kossnocorp) [inline src/nextMonday/index.ts:15]:
```suggestion
```

File: src/nextSaturday/index.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @name nextSaturday
+ * @category Weekday Helpers
+ * @summary When is the next Saturday?
+ *
+ * @description
+ * When is the next Saturday?
+ *
+ * @param {Date | number} date the date to check
```
Comment (kossnocorp) [inline src/nextSaturday/index.ts:13]:
```suggestion
 * @param {Date | number} date - the date to start counting from
```

File: src/nextSaturday/test.ts [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+      nextSaturday(new Date(2020, 2 /* Mar */, 18)),
+      new Date(2020, 2 /* Mar */, 21)
+    )
+
+    assert.deepStrictEqual(
+      nextSaturday(new Date(2020, 2 /* Mar */, 17)),
+      new Date(2020, 2 /* Mar */, 21)
+    )
+  })
+  it('returns `Invalid Date` if the given date is invalid', function () {
```
Comment (kossnocorp) [inline src/nextSaturday/test.ts:44]:
```suggestion
  })

  it('returns `Invalid Date` if the given date is invalid', () => {
```

File: src/nextSunday/index.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @name nextSunday
+ * @category Weekday Helpers
+ * @summary When is the next Sunday?
+ *
+ * @description
+ * When is the next Sunday?
+ *
+ * @param {Date | number} date the date to check
```
Comment (kossnocorp) [inline src/nextSunday/index.ts:13]:
```suggestion
 * @param {Date | number} date - the date to start counting from
```

File: src/nextSaturday/index.ts [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+ * @returns {Date} the date is the next Saturday
```
Comment (kossnocorp) [inline src/nextSaturday/index.ts:14]:
```suggestion
 * @returns {Date} the next Saturday
```

File: src/nextSunday/index.ts [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+ * @returns {Date} the date is the next Sunday
```
Comment (kossnocorp) [inline src/nextSunday/index.ts:14]:
```suggestion
 * @returns {Date} the next Sunday
```

File: src/nextSunday/index.ts [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+ * @throws {TypeError} the date should be a valid date
```
Comment (kossnocorp) [inline src/nextSunday/index.ts:15]:
```suggestion
```

File: src/nextSunday/index.ts [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+ * @throws {TypeError} 1 argument required
+ *
+ * @example
+ * When is the next Sunday after Mar, 22, 2020?
+ * const result = nextSunday(new Date(2020, 2, 22))
+ * => Sun Mar 29 2020 00:00:00
+ **/
+
+export default function nextSunday(date: Date | number): Date {
```
Comment (kossnocorp) [inline src/nextSunday/index.ts:24]:
```suggestion
 */
export default function nextSunday(date: Date | number): Date {
```

File: src/nextSaturday/index.ts (hunk 1 above)
Comment (kossnocorp) [inline src/nextSaturday/index.ts:22]:
```suggestion
 */
```

File: src/nextSunday/test.ts [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+      nextSunday(new Date(2020, 2 /* Mar */, 18)),
+      new Date(2020, 2 /* Mar */, 22)
+    )
+
+    assert.deepStrictEqual(
+      nextSunday(new Date(2020, 2 /* Mar */, 17)),
+      new Date(2020, 2 /* Mar */, 22)
+    )
+  })
+  it('returns `Invalid Date` if the given date is invalid', function () {
```
Comment (kossnocorp) [inline src/nextSunday/test.ts:44]:
```suggestion
  })

  it('returns `Invalid Date` if the given date is invalid', () => {
```

File: src/nextThursday/index.ts [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+ * @throws {TypeError} the date should be a valid date
+ * @throws {TypeError} 1 argument required
+ *
+ * @example
+ * When is the next Thursday after Mar, 22, 2020?
+ * const result = nextThursday(new Date(2020, 2, 22))
+ * => Thur Mar 26 2020 00:00:00
+ **/
+
+export default function nextThursday(date: Date | number): Date {
```
Comment (kossnocorp) [inline src/nextThursday/index.ts:24]:
```suggestion
 * @param {Date | number} date - the date to start counting from
 * @returns {Date} the next Thursday
 * @throws {TypeError} 1 argument required
 *
 * @example
 * When is the next Thursday after Mar, 22, 2020?
 * const result = nextThursday(new Date(2020, 2, 22))
 * => Thur Mar 26 2020 00:00:00
 */
export default function nextThursday(date: Date | number): Date {
```

File: src/nextThursday/test.ts [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+      nextThursday(new Date(2020, 2 /* Mar */, 18)),
+      new Date(2020, 2 /* Mar */, 19)
+    )
+
+    assert.deepStrictEqual(
+      nextThursday(new Date(2020, 2 /* Mar */, 17)),
+      new Date(2020, 2 /* Mar */, 19)
+    )
+  })
+  it('returns `Invalid Date` if the given date is invalid', function () {
```
Comment (kossnocorp) [inline src/nextThursday/test.ts:44]:
```suggestion
  })

  it('returns `Invalid Date` if the given date is invalid', () => {
```

File: src/nextTuesday/index.ts [hunk 22]
```diff
@@ -10,10 +10,10 @@ import toDate from '../toDate/index'
...
  * When is the next Tuesday?
  *
- * @param {Date | number} date - the date to check
- * @returns {Date} - the date is the next Tuesday
- * @throws {TypeError} - the date should be a valid date
- * @throws {TypeError} - 1 argument required
+ * @param {Date | number} date the date to check
+ * @returns {Date} the date is the next Tuesday
+ * @throws {TypeError} the date should be a valid date
+ * @throws {TypeError} 1 argument required
```
Comment (kossnocorp) [inline src/nextTuesday/index.ts:16]:
```suggestion
 * @param {Date | number} date - the date to start counting from
 * @returns {Date} the next Tuesday
 * @throws {TypeError} 1 argument required
```

File: src/nextWednesday/index.ts [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+ * @returns {Date} the date is the next Wednesday
+ * @throws {TypeError} the date should be a valid date
+ * @throws {TypeError} 1 argument required
+ *
+ * @example
+ * When is the next Wednesday after Mar, 22, 2020?
+ * const result = nextWednesday(new Date(2020, 2, 22))
+ * => Wed Mar 25 2020 00:00:00
+ **/
+export default function nextWednesday(date: Date | number): Date {
```
Comment (kossnocorp) [inline src/nextWednesday/index.ts:23]:
```suggestion
 * @param {Date | number} date - the date to start counting from
 * @returns {Date} the next Wednesday
 * @throws {TypeError} 1 argument required
 *
 * @example
 * When is the next Wednesday after Mar, 22, 2020?
 * const result = nextWednesday(new Date(2020, 2, 22))
 * => Wed Mar 25 2020 00:00:00
 */
export default function nextWednesday(date: Date | number): Date {
```

**Review by kossnocorp — CHANGES_REQUESTED** [review]

File: src/nextSaturday/index.ts (hunk 1 above)
Comment (tan75) [reply src/nextSaturday/index.ts:22]: Done.

**Review by kossnocorp — APPROVED** [review]
Thank you a lot 👏

**Pull request merged**